packages/fetch/src/types.ts

```diff
@@ -16,182 +16,174 @@
+  has(name: string): boolean;
+  set(name: string, value: string): void;
+  forEach(callbackfn: (value: string, key: string, parent: this) => void): void;
+  getSetCookie?(): string[];
+  keys?(): IterableIterator<string>;
+  values?(): IterableIterator<string>;
+  entries?(): IterableIterator<[string, string]>;
+  [Symbol.iterator]?(): Iterator<[string, string]>;
 }
 
-// AbortController
+export type Blob = unknown;
+export type FormData = unknown;
+
+export declare class Response<PlatformHeaders extends Headers = Headers> {
+  // constructor (body?: BodyInit, init?: ResponseInit);
 
-type AbortSignal = unknown;
+  readonly headers: PlatformHeaders;
+  readonly ok: boolean;
+  readonly status: number;
+  readonly statusText: string;
+  readonly type: ResponseType;
+  readonly url: string;
+  readonly redirected: boolean;
 
-/** A controller object that allows you to abort one or more DOM requests as and when desired. */
+  readonly body?: ReadableStream | null;
+  readonly bodyUsed: boolean;
 
-export type AbortController = {
-  /**
-   * The prototype of an instance is the class.
-   */
-  prototype: AbortController;
+  readonly arrayBuffer: () => Promise<ArrayBuffer>;
+  readonly blob: () => Promise<Blob>;
+  readonly formData: () => Promise<FormData>;
+  readonly json: () => Promise<unknown>;
+  readonly text: () => Promise<string>;
 
-  /**
-   * Constructs an AbortController.
-   */
-  new (): AbortController;
+  readonly clone: () => this;
 
-  /**
-   * Returns the AbortSignal object associated with this object.
-   */
-  readonly signal: AbortSignal;
-  /**
-   * Invoking this method will set this object's AbortSignal's aborted flag and signal to any observers that the associated activity is to be aborted.
-   */
-  abort(): void;
-};
+  /*
+  static error(): Response;
+  static json(data: any, init?: ResponseInit): Response;
+  static redirect(url: string | URL, status: ResponseRedirectStatus): Response;
+  */
+}
 
 /**
  * Minimal types for a ReadableStream.
  * @todo Missing declarations for `constructor`, `pipeThrough`, `pipeTo` and `tee`.
  */
 
-export type ReadableStream = {
+export type ReadableStream<T = unknown> = {
   /**
    * Creates a reader and locks the stream to it.
    * While the stream is locked, no other reader can be acquired until this one is released.
    * @see https://developer.mozilla.org/en-US/docs/Web/API/ReadableStream/getReader
    */
-  getReader(): StreamReader;
+  getReader(): ReadableStreamDefaultReader<T>;
+  getReader(options: { mode: "byob" }): unknown;
 
   /**
    * Cancel is used when you've completely finished with the stream and don't need any more data from it, even if there are chunks enqueued waiting to be read.
    * @return a Promise that resolves when the stream is canceled.
    * @see https://developer.mozilla.org/en-US/docs/Web/API/ReadableStream/cancel
    */
   cancel: (reason?: string) => Promise<void>;
-} & AsyncIterable<Uint8Array>;
+}; // & AsyncIterable<T>;
```

Comment:
```suggestion
};
```

Reply:
Again, this is explicitly left out (not simply forgotten), [if the Node types start implementing this](https://github.com/microsoft/TypeScript-DOM-lib-generator/issues/1677), we can add it back in - this comment might serve as a reminder to do just that.

Reply:
Perhaps an explicit comment would be helpful

Reply:
```suggestion
}; // & AsyncIterable<T>;
// Should be [AsyncIterable](https://developer.mozilla.org/en-US/docs/Web/API/ReadableStream#async_iteration), but not all our supported runtimes implement this.
```